cpp/include/cudf/detail/stream_compaction.hpp

```diff
@@ -82,10 +82,36 @@ std::unique_ptr<table> unique(
 std::unique_ptr<table> distinct(
   table_view const& input,
   std::vector<size_type> const& keys,
+  duplicate_keep_option keep          = duplicate_keep_option::KEEP_ANY,
   null_equality nulls_equal           = null_equality::EQUAL,
   rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Get the row indices that denote all distinct rows in `input`.
+ *
+ * Given an `input` table_view, each row index is copied to the output array in an
+ * unspecified order if the corresponding row of `keys` columns is distinct (no other
+ * equivalent row exists in the table).
+ *
+ * Note that the returning gather map is compact: all the indices are valid.
+ *
+ * @param input The input table_view to generate gather map for distinct rows
+ * @param keys Vector of indices representing key columns from `input`
+ * @param keep Keep the first, last, any, or no rows of the found duplicates
+ * @param nulls_equal Flag to specify whether null key elements should be considered as equal
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned table's device memory
+ * @return A device_uvector containing the resulting compact gather map
+ */
+rmm::device_uvector<size_type> get_distinct_indices(
+  table_view const& input,
+  std::vector<size_type> const& keys,
```

Comment:
This function should just take a `table_view` with the expectation that the caller will have already selected the relevant "key" columns if needed.